src/app/services/views/ServiceDetailView.vue

```diff
@@ -30,112 +30,133 @@
         v-slot="{ data, isLoading, error }: ServiceInsightSource"
         :src="`/meshes/${route.params.mesh}/service-insights/${route.params.service}`"
       >
+        <LoadingBlock v-if="isLoading" />
+
+        <ErrorBlock
+          v-else-if="error"
+          :error="error"
+        />
+
+        <EmptyBlock v-else-if="data === undefined" />
+
         <DataSource
+          v-else
           v-slot="{ data: externalService, isLoading: isLoadingExternalService, error: externalServiceError }: ExternalServiceSource"
+          :should-make-request="data.serviceType === 'external'"
           :src="`/meshes/${route.params.mesh}/external-services/${route.params.service}`"
         >
-          <LoadingBlock v-if="isLoading || isLoadingExternalService" />
+          <LoadingBlock v-if="data.serviceType === 'external' && isLoadingExternalService" />
 
           <ErrorBlock
-            v-else-if="error ?? externalServiceError"
-            :error="error ?? externalServiceError"
+            v-else-if="data.serviceType === 'external' && externalServiceError"
+            :error="externalServiceError"
           />
 
-          <EmptyBlock v-else-if="data === undefined || externalService === undefined" />
+          <EmptyBlock v-else-if="data.serviceType === 'external' && externalService === undefined" />
 
-          <template v-else>
-            <div class="stack">
+          <TabsWidget
```

Comment:
This is the current approach of how we establish a tabbed interface within detail views. I saw no reason to change this in the scope of this task because the current approach works. If we want to change change this approach, it’s separate work.

Reply:
Your call. I don’t think this has a high priority currently.

Reply:
Yeah but it just helps if everything works the same way, I'll make an issue, plus using tabs with query params is troublesome especially when we get all this for free with `vue-router` and nested routes.

Reply:
I made an issue for this https://github.com/kumahq/kuma-gui/issues/1331